Lbgpu does not compile with clang++ 4.0

(on icp)
[ 19%] Building CXX object src/core/CMakeFiles/EspressoCore.dir/lb.cpp.o
/home/weeber/espresso-dev/src/core/lb.cpp:785:24: error: non-constant-expression cannot be narrowed from type 'unsigned int' to 'int' in initializer list [-Wc++11-narrowing]
            bb_high = {lbpar_gpu.dim_x-1, lbpar_gpu.dim_y-1, lbpar_gpu.dim_z-1};
                       ^~~~~~~~~~~~~~~~~
/home/weeber/espresso-dev/src/core/lb.cpp:785:24: note: insert an explicit cast to silence this issue
            bb_high = {lbpar_gpu.dim_x-1, lbpar_gpu.dim_y-1, lbpar_gpu.dim_z-1};
                       ^~~~~~~~~~~~~~~~~
                       static_cast<int>()
/home/weeber/espresso-dev/src/core/lb.cpp:785:43: error: non-constant-expression cannot be narrowed from type 'unsigned int' to 'int' in initializer list [-Wc++11-narrowing]
            bb_high = {lbpar_gpu.dim_x-1, lbpar_gpu.dim_y-1, lbpar_gpu.dim_z-1};
                                          ^~~~~~~~~~~~~~~~~
/home/weeber/espresso-dev/src/core/lb.cpp:785:43: note: insert an explicit cast to silence this issue
            bb_high = {lbpar_gpu.dim_x-1, lbpar_gpu.dim_y-1, lbpar_gpu.dim_z-1};
                                          ^~~~~~~~~~~~~~~~~
                                          static_cast<int>()
/home/weeber/espresso-dev/src/core/lb.cpp:785:62: error: non-constant-expression cannot be narrowed from type 'unsigned int' to 'int' in initializer list [-Wc++11-narrowing]
            bb_high = {lbpar_gpu.dim_x-1, lbpar_gpu.dim_y-1, lbpar_gpu.dim_z-1};
                                                             ^~~~~~~~~~~~~~~~~
/home/weeber/espresso-dev/src/core/lb.cpp:785:62: note: insert an explicit cast to silence this issue
            bb_high = {lbpar_gpu.dim_x-1, lbpar_gpu.dim_y-1, lbpar_gpu.dim_z-1};
                                                             ^~~~~~~~~~~~~~~~~
                                                             static_cast<int>()
3 errors generated.

--- a/src/core/lb.cpp
+++ b/src/core/lb.cpp
@@ -782,7 +782,7 @@ int lb_lbfluid_print_vtk_velocity(char* filename, std::vector<int> bb1, std::vec
             bb_low = {0, 0, 0};
             if (lattice_switch & LATTICE_LB_GPU) {
 #ifdef LB_GPU
-            bb_high = {lbpar_gpu.dim_x-1, lbpar_gpu.dim_y-1, lbpar_gpu.dim_z-1};
+            bb_high = {static_cast<int>(lbpar_gpu.dim_x)-1, static_cast<int>(lbpar_gpu.dim_y)-1, static_cast<int>(lbpar_gpu.dim_z)-1};
 #endif // LB_GPU
             } else {
 #ifdef LB

--- a/src/core/particle_data.hpp
+++ b/src/core/particle_data.hpp
@@ -368,9 +368,9 @@ struct Particle {
   IntList &exclusions() {
 #ifdef EXCLUSIONS
     return el;
-#endif
-
+#else
     throw std::runtime_error{"Exclusions not enabled."};
+#endif
   }
 
 #ifdef EXCLUSIONS

--- a/src/core/scafacos.cpp
+++ b/src/core/scafacos.cpp
@@ -30,6 +30,7 @@
 #include <algorithm>
 #include <iostream>
 #include <limits>
+#include <numeric>
 
 #include "cells.hpp"
 #include "errorhandling.hpp"
@@ -222,6 +223,8 @@ double long_range_energy() {
 #endif	
     }
   }
+
+  return 0.0;
 }
 
 /** Determine runtime for a specific cutoff */

--- a/src/core/scafacos/Scafacos.cpp
+++ b/src/core/scafacos/Scafacos.cpp
@@ -147,8 +147,6 @@ void Scafacos::set_common_parameters(double *box_l, int *periodicity, int total_
    // Scafacos does near field calc
    sr=1;
   }
-  int d=dipolar();
-//  printf("Short range switch %d, dipolar=%d, has_near=%d\n",sr,d,has_near);
   handle_error(fcs_set_common(handle, sr, boxa, boxb, boxc, off, periodicity, total_particles));
 }
 

--- a/src/core/utils/List.hpp
+++ b/src/core/utils/List.hpp
@@ -56,7 +56,7 @@ public:
   T *end() { return e + n; }
   T const *end() const { return e + n; }
   size_type size() const { return n; }
-  bool empty() const { n == 0; }
+  bool empty() const { return n == 0; }
   size_type capacity() const { return max; }
   void clear() { resize(0); }
 

--- a/src/core/utils/Timer.cpp
+++ b/src/core/utils/Timer.cpp
@@ -40,7 +40,7 @@ Timer::get_all_stats() {
 
   Communication::mpiCallbacks().call(Timer::mpi_callback);
 
-  for (int i = 1; i < ret.size(); ++i) {
+  for (unsigned i = 1; i < ret.size(); ++i) {
     comm.recv(i, 42, ret[i]);
   }
 

--- a/src/core/utils/Timer.hpp
+++ b/src/core/utils/Timer.hpp
@@ -44,7 +44,7 @@ public:
 
     friend boost::serialization::access;
     template <class Archive>
-    void serialize(Archive &ar, const unsigned int version) {
+    void serialize(Archive &ar, const unsigned int /* version */) {
       ar &m_avg;
       ar &m_sig;
       ar &m_var;

--- a/src/python/espressomd/electrokinetics.pyx
+++ b/src/python/espressomd/electrokinetics.pyx
@@ -6,8 +6,7 @@ from . import utils
 import numpy as np
 
 IF ELECTROKINETICS:
-
-    cdef class Electrokinetics(lb.HydrodynamicInteraction):
+    cdef class Electrokinetics(HydrodynamicInteraction):
         species_list = []
 
         def __getitem__(self, key):

